packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,292 @@
+            icon={KebabHorizontalIcon}
+            className={classes.MenuButton}
+            {...rest}
+          />
+        </ActionMenu.Anchor>
+        <ActionMenu.Overlay width="auto">
+          <ActionList role="menu">
+            {items.map((item, index) => {
+              const href = item.props.href
+              const children = item.props.children
+              const selected = item.props.selected
+              return (
+                <ActionList.LinkItem
+                  key={index}
+                  href={href}
+                  role="menuitem"
+                  aria-current={selected ? 'page' : undefined}
+                >
+                  {children}
+                </ActionList.LinkItem>
+              )
+            })}
+          </ActionList>
+        </ActionMenu.Overlay>
+      </ActionMenu>
+    )
+  },
+)
+
+BreadcrumbsMenuItem.displayName = 'Breadcrumbs.MenuItem'
+
+const getValidChildren = (children: React.ReactNode) => {
+  return React.Children.toArray(children).filter(child => React.isValidElement(child)) as React.ReactElement[]
+}
+
+function Breadcrumbs({className, children, sx: sxProp, overflow = 'wrap', hideRoot = true}: BreadcrumbsProps) {
+  const containerRef = useRef<HTMLElement>(null)
+  const menuButtonRef = useRef<HTMLButtonElement>(null)
+  const [effectiveHideRoot, setEffectiveHideRoot] = useState<boolean>(hideRoot)
+  //let effectiveOverflow = 'wrap'
+  const childArray = useMemo(() => getValidChildren(children), [children])
+
+  const rootItem = childArray[0]
+
+  const [visibleItems, setVisibleItems] = useState<React.ReactElement[]>(() => childArray)
+  const [childArrayWidths, setChildArrayWidths] = useState<number[]>(() => [])
+
+  const [menuItems, setMenuItems] = useState<React.ReactElement[]>([])
+  const [rootItemWidth, setRootItemWidth] = useState<number>(0)
+
+  // Menu button width with fallback
+  const MENU_BUTTON_FALLBACK_WIDTH = 32 // Design system small IconButton
+  const [menuButtonWidth, setMenuButtonWidth] = useState(MENU_BUTTON_FALLBACK_WIDTH)
+
+  // if (typeof window !== 'undefined') {
+  //   effectiveOverflow = overflow
+  // }
+  const MIN_VISIBLE_ITEMS = !effectiveHideRoot ? 3 : 4
+
+  useEffect(() => {
+    const listElement = containerRef.current?.querySelector('ol')
+    if (listElement && listElement.children.length > 0) {
+      const listElementArray = Array.from(listElement.children) as HTMLElement[]
+      const widths = listElementArray.map(child => child.offsetWidth)
+      setChildArrayWidths(widths)
+      setRootItemWidth(listElementArray[0].offsetWidth)
+    }
+  }, [childArray.length])
+
+  // Measure actual menu button width when it exists
+  useEffect(() => {
+    if (menuButtonRef.current) {
+      const measuredWidth = menuButtonRef.current.offsetWidth
+      if (measuredWidth > 0) {
+        setMenuButtonWidth(measuredWidth)
+      }
+    }
+  }, [menuItems.length]) // Re-measure when menu button appears/disappears
```

Comment:
One idea for measuring the menu button could be to measure it in a ref callback, curious if that would make sense or not 🤔 Would avoid the need for us to have an effect for it. Hopefully this doesn't violate the rules of hooks too 😅 lol